processing/src/main/java/io/druid/query/metadata/SegmentMetadataQueryQueryToolChest.java

```diff
@@ -284,4 +242,73 @@ public boolean apply(T input)
         )
     );
   }
+
+  @VisibleForTesting
+  public static SegmentAnalysis mergeAnalyses(final SegmentAnalysis arg1, final SegmentAnalysis arg2)
+  {
+    if (arg1 == null) {
+      return arg2;
+    }
+
+    if (arg2 == null) {
+      return arg1;
+    }
+
+    List<Interval> newIntervals = null;
+    if (arg1.getIntervals() != null) {
+      newIntervals = Lists.newArrayList();
+      newIntervals.addAll(arg1.getIntervals());
+    }
+    if (arg2.getIntervals() != null) {
+      if (newIntervals == null) {
+        newIntervals = Lists.newArrayList();
+      }
+      newIntervals.addAll(arg2.getIntervals());
+    }
+
+    final Map<String, ColumnAnalysis> leftColumns = arg1.getColumns();
+    final Map<String, ColumnAnalysis> rightColumns = arg2.getColumns();
+    Map<String, ColumnAnalysis> columns = Maps.newTreeMap();
+
+    Set<String> rightColumnNames = Sets.newHashSet(rightColumns.keySet());
+    for (Map.Entry<String, ColumnAnalysis> entry : leftColumns.entrySet()) {
+      final String columnName = entry.getKey();
+      columns.put(columnName, entry.getValue().fold(rightColumns.get(columnName)));
+      rightColumnNames.remove(columnName);
+    }
+
+    for (String columnName : rightColumnNames) {
+      columns.put(columnName, rightColumns.get(columnName));
+    }
+
+    final List<AggregatorFactory[]> aggregatorss = Lists.newArrayList();
+    if (arg1.getAggregators() != null) {
+      aggregatorss.add(arg1.getAggregators());
+    }
+    if (arg2.getAggregators() != null) {
+      aggregatorss.add(arg2.getAggregators());
+    }
+
+    return new SegmentAnalysis(
+        "merged",
+        newIntervals,
+        columns,
+        arg1.getSize() + arg2.getSize(),
+        arg1.getNumRows() + arg2.getNumRows(),
```

Comment:
Different rows in different segments count as two rows, so there can't be overlap